subsystems/viamserver/viamserver.go

```diff
@@ -304,15 +309,25 @@ func (s *viamServer) isRestartAllowed(ctx context.Context) (bool, error) {
 				resultChan <- nil
 				return
 			}
-			resultChan <- errors.New("viam-server reports it is unsafe to restart")
+			resultChan <- errUnsafeToRestart
 		}()
 	}
+
 	var combinedErr error
 	for i := 1; i <= len(urls); i++ {
 		result := <-resultChan
+
+		// If any test URL reports it is explicitly _safe_ to restart (nil value sent to
+		// resultChan), we can assume we can restart viam-server.
```

Comment:
Correct me if I'm wrong, but this comment is meant to describe the current behavior of testing against all "test URLs" here.